modules/monitoring-configuring-remote-write-storage.adoc

```diff
@@ -88,27 +116,93 @@ data:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote-write.endpoint"
+      - url: "https://basicAuth.remotewrite.com/api/write"
+        basicAuth:
+          username:
+            name: rw-auth
+            key: <username>
+          password:
+            name: rw-auth
+            key: <password>
+      - url: "https://authorization.remotewrite.com/api/write"
+        authorization:
+          type: Bearer
+          credentials:
+            name: rw-auth
+            key: <token_name>
 ----
+* OAuth 2.0
++
+An OAuth 2.0 (`oauth2`) configuration uses the client credentials grant type. Prometheus fetches an access token from the specified remote write endpoint with the specified client access and secret keys.
+You cannot use this method simultaneously with `basic_auth`, `Bearer`, or `sigv2` authentication.
 +
-* The following example configures client TLS authentication:
+The following shows an `oauth2` authentication sample configuration.
+It assumes you have already created an `oauth2` secret with the name `oauth2-credentials` in the `openshift-monitoring` namespace.
+Replace the sample `key`, `scopes`, and `endpointParams` values with your authentication credentials and other values.
 +
 [source,yaml]
 ----
-tlsConfig:
-  ca:
-    <caSecret>
-  cert:
-    <certSecret>
-  keySecret:
-    <keySecret>
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+        - url: https://test.remotewrite.com/api/write
+          remoteTimeout: 30s
+          oauth2:
+            clientId:
+              secret:
+                name: oauth2-credentials
+                key: <id>
+            clientSecret:
+              name: oauth2-credentials
+              key: <secret>
+            tokenUrl: https://example.com/oauth2/token
+            scopes:
+              - <scope_1>
+              - <scope_2>
+            endpointParams:
+              param1: <value_1>
+              param2: <value_2>
 ----
-Substitute `<caSecret>`, `<certSecret>`, and `<keySecret>` accordingly. 
 +
-The following sample shows a TLS authentication configuration using `selfsigned-mtls-bundle` for the `name` values and `ca.crt` for the `ca` `key` value, `client.crt` for the `cert` `key` value, and `client.key` for the `keySecret` `key` value:
+* TLS client
++
+A TLS client (`tlsConfig`) configuration specifies the CA certificate, the client certificate, and the client key file information used to authenticate with the remote write endpoint server using TLS.
++
+The following sample shows a `tlsConfig` configuration.
+It assumes that you have already created a CA certificate file, a client certificate file, and a client key file.
```

Comment:
It assumes you have already created a CA certificate file, a client certificate file, and a client key file. You also have created a `tls` secret with the name `selfsigned-mtls-bundle` in the `openshift-monitoring` namespace.

Reply:
Added the new sentence